src/Datadog.Trace/DiagnosticListeners/AspNetCoreDiagnosticObserver.cs

```diff
@@ -418,6 +872,48 @@ public struct BadHttpRequestExceptionStruct
             public int StatusCode;
         }
 
+        [DuckCopy]
+        public struct HttpRequestInEndpointMatchedStruct
+        {
+            [Duck(BindingFlags = DuckAttribute.DefaultFlags | BindingFlags.IgnoreCase)]
+            public HttpContext HttpContext;
+        }
+
+        [DuckCopy]
+        public struct EndpointFeatureStruct
+        {
+            public RouteEndpoint Endpoint;
+        }
+
+        [DuckCopy]
+        public struct HttpRequestStruct
+        {
+            public string Method;
+            public RouteValueDictionary RouteValues;
+            public PathString PathBase;
+        }
+
+        [DuckCopy]
+        public struct RoutePatternPathSegmentStruct
+        {
+            public IEnumerable Parts;
+        }
+
+        [DuckCopy]
+        public struct RoutePatternContentPartStruct
```

Comment:
What type is this proxying? I had trouble finding out the type from ASP.NET Core docs

Reply:
Ah yes, it's actually being used to proxy two different types `RoutePatternLiteralPart` and `RoutePatternSeparatorPart`, as they both expose the `Content` property we need. I'll add doc comments to the types to make it clear.

However, I wonder, is proxying two types like this sensible, or should we have a separate type for each? I assumed this was preferable, as it reduces the number of casts we have to do, and the number of duck types in general 🤔

Reply:
Actually this is completely fine, thanks for documenting the two types. When we attempt the duck type for each underlying instance, it will succeed because the `Content` property is present, and we'll just have two `Type`->`DuckTypeProxy` entries in our cache 👍🏼